lib/Echidna/Solidity.hs

```diff
@@ -108,10 +108,36 @@ data SolConf = SolConf { _contractAddr    :: Addr             -- ^ Contract addr
                        }
 makeLenses ''SolConf
 
+-- | List of contract names from every source cache
+type SourceCaches = [([ContractName], SourceCache)]
```

Comment:
wouldn't a Map be better here? or at least non-empty list?

Reply:
I tried that, but it is only used once, and we need `find` because of the signature that takes a boolean function (believe it or not, there is no equivalent in `Data.Map`. Same with NE lists). This will probably be refactored in later commit.